### PR TITLE
Extensions for Locale

### PR DESCRIPTION
**Pull request by jitinsharma**
#310

### CODE REVIEW

**Comment by jitinsharma** [comment]
I signed it!

File: src/androidTest/java/androidx/util/LocaleTest.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.util
+
+import android.view.View
+import junit.framework.Assert.assertEquals
```
Comment (JakeWharton) [inline src/androidTest/java/androidx/util/LocaleTest.kt]: nit: use `org.junit.Assert.assertEquals`

File: src/main/java/androidx/util/Locale.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import android.support.annotation.RequiresApi
+import android.text.TextUtils
+import java.util.Locale
+
+/**
+ * Returns layout direction for a given locale.
+ */
+@RequiresApi(17)
+fun Locale.getLayoutDirection(): Int = TextUtils.getLayoutDirectionFromLocale(this)
```
Comment (JakeWharton) [inline src/main/java/androidx/util/Locale.kt]: This seems like it should be a `val` and not a `fun`.

File: src/main/java/androidx/util/Locale.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+package androidx.util
+
+import android.support.annotation.RequiresApi
+import android.text.TextUtils
+import java.util.Locale
+
+/**
+ * Returns layout direction for a given locale.
+ */
```
Comment (JakeWharton) [inline src/main/java/androidx/util/Locale.kt]: Can you add `@see TextUtils.getLayoutDirectionFromLocale` to this?

**Comment by jitinsharma** [comment]
Updated the PR.
Apparently there was no change in current.txt after running ./gradlew updateApi

File: src/main/java/androidx/util/Locale.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import android.support.annotation.RequiresApi
+import android.text.TextUtils
+import java.util.Locale
+
+/**
+ * Returns layout direction for a given locale.
+ * @see TextUtils.getLayoutDirectionFromLocale
+ */
+
```
Comment (romainguy) [inline src/main/java/androidx/util/Locale.kt]: Remove empty line

File: src/androidTest/java/androidx/util/LocaleTest.kt [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+package androidx.util
+
+import android.view.View
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import java.util.Locale
+
+class LocaleTest {
+
```
Comment (romainguy) [inline src/androidTest/java/androidx/util/LocaleTest.kt]: Remove empty line

Comment (jitinsharma) [reply src/androidTest/java/androidx/util/LocaleTest.kt]: Done

File: src/main/java/androidx/util/Locale.kt (hunk 4 above)
Comment (jitinsharma) [reply src/main/java/androidx/util/Locale.kt]: Done

**Comment by romainguy** [comment]
Compiled & ran the tests locally.

**Pull request merged**